### PR TITLE
Generate header ids for each level

### PR DESCRIPTION
**Pull request by Mousticke**
## Description

The generate-heading-ids script does not go deeper than h3s. In md, the max level of heading is 6. 
But the fixed version of the script will check the maximum level and generate the header number without useless 0s for heading level not used.

## Related Issue

Related to [generate-heading-ids script is a bit broken #6529 ](https://github.com/ethereum/ethereum-org-website/issues/6529)

### CODE REVIEW

File: src/content/community/research/index.md [hunk 1]
```diff
@@ -5,393 +5,439 @@ sidebar: true
 lang: en
 ---
 
-# Active areas of Ethereum research
+# Active areas of Ethereum research {#active-areas-of-ethereum-research}
 
-One of the primary strengths of Ethereum is that an active research and engineering community are constantly improving it. Many enthusiastic, skilled people worldwide would like to apply themselves to outstanding issues in Ethereum, but it is not always easy to find out what those issues are. This page outlines key active research areas as a rough guide to Ethereum's cutting edge.
+One of the primary strengths of Ethereum is that it is constantly being improved by an active research and engineering community. There are many enthusiastic, skilled people worldwide that would like to apply themselves to outstanding issues in Ethereum, but it is not always easy to find out what those issues are. This page aims to outline key areas of active research as a rough guide to Ethereum's cutting edge.
```
Comment (samajammin) [inline src/content/community/research/index.md:10]: Not sure why these copy changes are in here - perhaps the file is outdated? Could you please sync this up with latest in `dev` branch, then re-apply the header ids?

Comment (samajammin) [inline src/content/community/research/index.md:10]: Not sure why these copy changes are in here - perhaps the file is outdated? Could you please sync this up with latest in `dev` branch, then re-apply the header ids?

File: src/scripts/generate-heading-ids.js [hunk 2]
```diff
@@ -57,10 +59,22 @@ const addHeaderID = (line, slugger, write = false) => {
     .replace(/\{#[^}]+\}/, "")
     .trim()
   const headingLevel = line.slice(0, line.indexOf(" "))
+
+  curMaxLevel =
+    headingLevel.length > curMaxLevel ? headingLevel.length : curMaxLevel
+
+  const headingLevelArrayIndex = headingLevel.length - 1
+  if (
+    curLevel[headingLevelArrayIndex] == undefined ||
```
Comment (samajammin) [inline src/scripts/generate-heading-ids.js:68]: May be able to simplify this statement by just checking if it's a truthy value or not, i.e. 
```
  if (!curLevel[headingLevelArrayIndex]) {
    curLevel[headingLevelArrayIndex] = 0
  }
```

File: src/scripts/generate-heading-ids.js [hunk 3]
```diff
@@ -57,10 +59,22 @@ const addHeaderID = (line, slugger, write = false) => {
...
+    curLevel[headingLevelArrayIndex] == null
+  ) {
+    curLevel[headingLevelArrayIndex] = 0
+  }
   curLevel[headingLevel.length - 1]++
```
Comment (samajammin) [inline src/scripts/generate-heading-ids.js:73]:
```suggestion
  curLevel[headingLevelArrayIndex]++
```
Should we use `headingLevelArrayIndex` her now that variable exists?

**Review by samajammin — CHANGES_REQUESTED** [review]
Awesome, thanks for the PR! Code mostly looks good & I tested the script locally on a file - looks to work 👍👍 

Left a couple suggestions on the script. Also see my comment re: the markdown file content - I don't think we should actually be changing any content as part of this PR.

File: src/content/community/research/index.md (hunk 1 above)
Comment (Mousticke) [reply src/content/community/research/index.md]: By the time, I took the file from the example you gave. #6527 :)

File: src/scripts/generate-heading-ids.js (hunk 2 above)
Comment (Mousticke) [reply src/scripts/generate-heading-ids.js]: Of course it's better 👍🏽

File: src/scripts/generate-heading-ids.js (hunk 3 above)
Comment (Mousticke) [reply src/scripts/generate-heading-ids.js]: Indeed we should

File: src/content/community/research/index.md (hunk 1 above)
Comment (samajammin) [reply src/content/community/research/index.md]: Ah, gotcha. Makes sense. Always worth ensuring your files are up to date with the latest `dev` when you open a PR 👍 

Thanks for fixing!

**Review by samajammin — APPROVED** [review]
Amazing - thanks for the updates! 🎉

**Pull request merged**